code/game/objects/items.dm

```diff
@@ -53,6 +52,10 @@
 	var/vending_cat = null// subcategory for vending machines.
 	var/list/dynamic_overlay[0] //For items which need to slightly alter their on-mob appearance while being worn.
 
+	var/shrapnel_amount = 0 // How many pieces of shrapnel it disintegrates into. This variable also decides it the item disintegrates or not!
```

Comment:
>Wouldn't I need to define the proc separately for every item i want to be shrapnel then?

You could make it an item-level proc that returns 0 by default, and then just override it for the things you want to be shrapnel.